### DISCUSSION TITLE
How does the Windows community toolkit gallery app persists its previous window location?

### DISCUSSION THREAD

**Discussion by aries-zhang** [CommunityToolkit/Windows#524] [Q&A] [answered]
It looks like the [Windows Community Toolkit Gallery](https://apps.microsoft.com/detail/9nblggh4tlcq) app is able to keep its previous window size and location when relaunched. Does anyone know how it is achieved in WinUI 3, especially across multiple screens and when these screens do not have same DPI settings? I cloned this repo but didn't seem to find any implementation around this. Much appreciated if someone can point me to the right location in the code.

The background I came from is that WinUI 3 does not seem to provide a set of handy APIs like `System.Windows.Forms.Screen` or `System.Windows.Media.VisualTreeHelper.GetDpi`. Although there are Win32 APIs such as [GetWindowPlacement](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getwindowplacement) and [GetMonitorInfo](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getmonitorinfoa), but wondering if there is another way that Community Toolkit Gallery app achieves so since it does not seem to call these APIs?

**Comment by niels9001** [comment, marked as answer]
The Gallery app from the Store is still using UWP, which comes with this behavior out of the box.

IIRC WinUiEx has a few helpers to do this for WinUI apps.

**Reply by aries-zhang** [reply]
Thanks @niels9001 👍
